### ISSUE TITLE
Birthday card mailings

### ISSUE THREAD

**Issue by brotzmanmj** [episphere/connect#736]
NORC will be sending birthday cards via snail mail on a monthly basis to our participants in their birth month. 

They will pull the information via API for all participants who are **verified** and **not withdrawn** and have birthdays in a given month.

The variables they will need are:
- Connect ID
- Name (current first and last name from User Profile)
- Mailing address (current address from User Profile)

Do not include the date of birth because we do not want to proliferate PII into their system. NORC will make the API call once a month, and we need to populate the list with people whose birthday is that month.

NORC will send back to us via API:
- Disposition Code (NORC will define and will also let us know when they will push this and if it will be static or updated)
- Updated address if NORC receives back an updated address from the post office. We will want this mailing address to update the address fields in the User Profile and retain address history in the same way all post-verification user profile updates do if made through the PWA or SMDB but in this case, will be via API.

**Comment by brotzmanmj** [comment]
Note from 10/16/2023 meeting with NORC: For launch, we will move forward without the requirement that NORC send data back to us. It is expected to be at least 2-3 months before data will be available to send back. That should become a separate item.

**Comment by Davinkjohnson** [comment]
Current technical plan: 
Add a filter to the "filters" section of the getParticipantsData for "Birthday Month", pass in a value 1-12 for the month.
Do not filter off deceased participants on our end, have the NORC team remove/filter these participants on their end.

**Comment by brotzmanmj** [comment]
Will the NORC team also remove people who have withdrawn from their end?

**Comment by Davinkjohnson** [comment]
@anthonypetersen We didn't talk about the Withdrawn criteria. Can that be handled as part of the getParticipants call currently or would there need to be a change made to enable that?

**Comment by anthonypetersen** [comment]
@Davinkjohnson looking at our code, if you use the `filter` option for the `getParticipants` API the only additional filter we are adding on in our code is that the participants returned belong to the site that made the call, unless you are NORC and you will see everyone. So we aren't doing anything at the moment to filter off withdrawn or unverified participants.

**Comment by anthonypetersen** [comment]
@brotzmanmj @Davinkjohnson I think it would be a good idea to add filters on the API end that automatically filters off people who are unverified or withdrawn if you are using the `filter` option for this API

**Comment by brotzmanmj** [comment]
@anthonypetersen I could be mistaken but don't we already have a couple of filter options on the API for different purposes?

**Comment by anthonypetersen** [comment]
@brotzmanmj yes but you can't combine them, so if we added an additional filter for birth month you can't combine it with the verified option

**Comment by brotzmanmj** [comment]
@anthonypetersen  I see, so you're suggesting for the filtered API that is specifically for the birthday cards, in addition to limiting by birth month it would filter out anyone with verification status not 'verified' and anyone withdrawn? But not deceased?

**Comment by anthonypetersen** [comment]
@brotzmanmj I'm saying that anytime the `filter` option is used we automatically filter out people that aren't verified or have withdrawn

**Comment by brotzmanmj** [comment]
So I don't understand what the implications of that will be for the sites ongoing operations

**Comment by anthonypetersen** [comment]
@brotzmanmj If you use the filter option you can search by the following things:

* first name
* last name
* email
* phone number 
* dob
* connect ID
* studyId

what we'd be saying is "if you're going to filter our database of participants for these things, you're only going to get results of verified participants that haven't withdrawn from the study"

**Comment by brotzmanmj** [comment]
so what I don't know is what impact that would have on the sites. Could we discuss tomorrow and maybe include Nicole? thanks

**Comment by anthonypetersen** [comment]
Will add baseline functionality for upcoming release. Decision was made to query sites as to how they are using the `filter` option on this API and to let them know modifications will be coming.

**Comment by anthonypetersen** [comment]
@ramavavi and @abhinavjonnada82 have both reviewed and approved

**Comment by anthonypetersen** [comment]
tested & approved in dev

**Comment by brotzmanmj** [comment]
Is this ready for NORC to test in dev?

**Comment by anthonypetersen** [comment]
Yup, they just need to use the option `birthMonth' when using the `filter` option on the API

**Comment by brotzmanmj** [comment]
Thanks Tony! @mnataraj92 can you reach out to NORC to begin testing?

**Comment by anthonypetersen** [comment]
NORC to begin testing Monday

**Comment by FrogGirl1123** [comment]
Once NORC is ready to send back variables, loop in Analytics team to create needed variables, check back end, and QC and Metrics if needed.

**Comment by brotzmanmj** [comment]
@FrogGirl1123 thanks for the reminder. Since it will be a couple of months before we establish what NORC will send back and create variables for that, is it OK if I open a separate issue for it and put it on hold, so we can close this issue with the current release?

**Comment by sonyekere** [comment]
Pending add'l feedback, Dec release pending.

**Comment by FrogGirl1123** [comment]
As decided through email and Punch List conversations on Wed., Nov. 29, we will put a hold on development of the API filter. Instead we will create a csv file with the with the following requirements to be shared with NORC via Box as we do for the BQ1 data for the sites:
For all participants who are verified and not withdrawn and have birthdays in a given month (e.g. January for out first shared file).
The csv file will contain:
Connect ID
Name (current first and last name from User Profile)
Mailing address (current address from User Profile)

@brotzmanmj please add any additional requirements and let @mnataraj92 where to set up the folder for NORC. Also what would we like as the delivery date, first day of the month, end of the previous month?

@brotzmanmj, @sonyekere, @anthonypetersen  should we break this into a new Issue since the API work will resume eventually? 

@jeannewu I would like you to write the scheduled query for this table and @jacobmpeters to work on getting the resulting csv into Box

**Comment by jeannewu** [comment]
@FrogGirl1123 Thanks a lot. I will go over the data and try to get the sql for this csv file. I will let you know if I have any questions.

**Comment by brotzmanmj** [comment]
add condition: And not deceased
thanks

**Comment by jeannewu** [comment]
@ALL, just double check on the request of the birthday card in NORC. All the relevant info are all from the PII data in BQ1, right?

**Comment by FrogGirl1123** [comment]
> @ALL, just double check on the request of the birthday card in NORC. All the relevant info are all from the PII data in BQ1, right?

Yes, @jeannewu the PII/csv file will come from BQ1

**Comment by JoeArmani** [comment]
@anthonypetersen 
I’ve been reviewing this issue and the CSV issue: https://github.com/episphere/connect/issues/831.

Here's a best guess at Initial Requirements (Please comment/correct/update where needed):

**Query Participants Where:**
•Birth Month query match (564964481 == range of values 01 - 12 with leading zeros) 
•Verified == yes (821247024 == 197316935)
•Withdrawn == no (747006172 == 104430631)
•deceased == no (987563196 == 104430631)


**Return Data For Each Participant (match current CSV Values with JSON response):**
•CONNECT_ID
•PIN (same as CONNECT_ID)
•DOBM (two-digit with leading zeros)
•first_name
•last_name
•address_line_1
•address_line_2
•city
•state
•zip_code


**Questions:**
(1) Do we want to create a new ‘getBirthdays’ API, specialized for this use case or use an existing API? A new API might help ease the confusion we’ve recently discussed in the getParticipants API. In this case, NORC could just submit the `birthMonth` and `page` values in a query to the API and everything else would be handled on the back end. Either option works well on my end.

(2) Is site code a factor here, or does this apply to all participants study-wide? (I think all participants, but wanted to confirm).

(3) I noticed ‘UP Suffix’ 506826178 is not included. Do we want to include this field? We can concatenate this to the ‘last_name’ field when it exists, provide it as a separate ‘suffix’ field, or omit it.

Test data from dev - this matches the current CSV file headings (the actual response would be a list of participants, not a single participant):
<img width="500" alt="Screenshot 2024-02-06 at 1 57 06 PM" src="https://github.com/episphere/connect/assets/93854858/0a65f271-2180-4433-bb63-72f4743bbb01">
<img width="454" alt="Screenshot 2024-02-06 at 2 00 10 PM" src="https://github.com/episphere/connect/assets/93854858/023e132a-022f-435a-953f-92519298d023">

**Comment by sonyekere** [comment]
Closing this issue. A new issue will be created to address a different request.
